central/processlisteningonport/datastore/datastore_impl_test.go

```diff
@@ -0,0 +1,1099 @@
+package datastore
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	processIndicatorDataStore "github.com/stackrox/rox/central/processindicator/datastore"
+	processIndicatorSearch "github.com/stackrox/rox/central/processindicator/search"
+	processIndicatorStorage "github.com/stackrox/rox/central/processindicator/store/postgres"
+	plopStore "github.com/stackrox/rox/central/processlisteningonport/store/postgres"
+	postgresStore "github.com/stackrox/rox/central/processlisteningonport/store/postgres"
+	"github.com/stackrox/rox/central/role/resources"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/fixtures/fixtureconsts"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestPLOPDataStore(t *testing.T) {
+	suite.Run(t, new(PLOPDataStoreTestSuite))
+}
+
+type PLOPDataStoreTestSuite struct {
+	suite.Suite
+	datastore          DataStore
+	store              plopStore.Store
+	indicatorDataStore processIndicatorDataStore.DataStore
+
+	postgres *pgtest.TestPostgres
+
+	hasNoneCtx  context.Context
+	hasReadCtx  context.Context
+	hasWriteCtx context.Context
+
+	mockCtrl *gomock.Controller
+}
+
+func (suite *PLOPDataStoreTestSuite) SetupSuite() {
+	if !env.PostgresDatastoreEnabled.BooleanSetting() {
+		suite.T().Skip("Skip PLOP tests if postgres is disabled")
+		suite.T().SkipNow()
+	}
+	suite.hasNoneCtx = sac.WithGlobalAccessScopeChecker(context.Background(), sac.DenyAllAccessScopeChecker())
+	suite.hasReadCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.DeploymentExtension)))
+	suite.hasWriteCtx = sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.DeploymentExtension)))
+}
+
+func (suite *PLOPDataStoreTestSuite) SetupTest() {
```

Comment:
I appreciate the better consolidation of suite setup.
Using a new DB for each test will be good for parallelization.